akka-docs/src/main/paradox/scala/camel.md

```diff
@@ -169,73 +224,122 @@ and the actor replies to the endpoint when the response is ready. The ask reques
+@scala[@extref[Producer](github:akka-camel/src/main/scala/akka/camel/Producer.scala).transformOutgoingMessage methods]@java[@extref[UntypedProducerActor](github:akka-camel/src/main/scala/akka/camel/javaapi/UntypedProducerActor.scala).onTransformOutgoingMessage methods]. This is explained later in more detail.
+Producer Actors cannot override the @scala[default @extref[Producer](github:akka-camel/src/main/scala/akka/camel/Producer.scala).receive]@java[@extref[UntypedProducerActor](github:akka-camel/src/main/scala/akka/camel/javaapi/UntypedProducerActor.scala).onReceive] method.
+
+Any message sent to a @scala[@extref[`Producer`](github:akka-camel/src/main/scala/akka/camel/Producer.scala)]@java[Producer] actor will be sent to
 the associated Camel endpoint, in the above example to
-`http://localhost:8080/news`. The @extref[Producer](github:akka-camel/src/main/scala/akka/camel/Producer.scala) always sends messages asynchronously. Response messages (if supported by the
+`http://localhost:8080/news`. The @scala[@extref[`Producer`](github:akka-camel/src/main/scala/akka/camel/Producer.scala)]@java[@extref[`UntypedProducerActor`](github:akka-camel/src/main/scala/akka/camel/javaapi/UntypedProducerActor.scala)] always sends messages asynchronously. Response messages (if supported by the
 configured endpoint) will, by default, be returned to the original sender. The
 following example uses the ask pattern to send a message to a
 Producer actor and waits for a response.
 
-@@snip [Producers.scala]($code$/scala/docs/camel/Producers.scala) { #AskProducer }
+Scala
+:  @@snip [Producers.scala]($code$/scala/docs/camel/Producers.scala) { #AskProducer }
+
+Java
+:  @@snip [ProducerTestBase.java]($code$/java/jdocs/camel/ProducerTestBase.java) { #AskProducer }
 
-The future contains the response CamelMessage, or an `AkkaCamelException` when an error occurred, which contains the headers of the response.
+The future contains the response `CamelMessage`, or an `AkkaCamelException` when an error occurred, which contains the headers of the response.
 
 <a id="camel-custom-processing"></a>
 ### Custom Processing
 
 Instead of replying to the initial sender, producer actors can implement custom
-response processing by overriding the routeResponse method. In the following example, the response
+response processing by overriding the @scala[`routeResponse`]@java[`onRouteResponse`] method. In the following example, the response
 message is forwarded to a target actor instead of being replied to the original
 sender.
 
+@@@ div { .group-scala }
+
 @@snip [Producers.scala]($code$/scala/docs/camel/Producers.scala) { #RouteResponse }
 
+@@@ 
+
+@@@ div { .group-java }
+
+@@snip [ResponseReceiver.java]($code$/java/jdocs/camel/ResponseReceiver.java) { #RouteResponse }
```

Comment:
Three code snippets in different files, so we cannot use tabs to separate Java/Scala, but have to enclose them in `@@@ div { .group-java }`